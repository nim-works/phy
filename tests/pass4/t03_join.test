discard """
  description: "Renaming back-propagates where possible"
"""
(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) (Continuations
    (Continuation (Params) (Locals)
      (SelectBool (IntVal 0)
        (Continue 1 (List))
        (Continue 2 (List)))
    )
    (Continuation (Params)
      (Locals (Type 0))
      (Continue 3
        (List (Move (Local 0))))
    )
    (Continuation (Params)
      (Locals (Type 0))
      (Continue 3
        (List (Move (Local 0))))
    )
    (Continuation (Params (Type 0)) (Locals)
      (Continue 4 (List))
    )
    (Continuation (Params)))
  )
)
