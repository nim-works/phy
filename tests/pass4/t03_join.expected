;$sexp
(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1)
    (Locals (Type 0))
    (Continuations
      (Continuation (Params) (Locals)
        (SelectBool (IntVal 0)
          (Continue 1)
          (Continue 2)
        )
      )
      (Continuation (Params)
        (Locals (Local 0))
        (Continue 3)
      )
      (Continuation (Params)
        (Locals (Local 0))
        (Continue 3)
      )
      (Continuation (Params)
        (Locals (Local 0))
        (Continue 4)
      )
      (Continuation (Params))
    )
  )
)
